### ISSUE TITLE
RF BatchNorm running var small diff between TF-layers, pure RF and direct PyTorch, biased vs unbiased

### ISSUE THREAD

**Issue by albertz** [rwth-i6/returnn#1539]
There are multiple implementations of batch norm, but here, three different cases are relevant:

1. The pure RF implementation (which is used e.g. when `use_mask=True`)
2. RF TF-layers backend, via `LayerBase.batch_norm`
3. RF PyTorch backend, which uses `torch.nn.functional.batch_norm`

In case 1 and 2, we use the biased estimate of the batch variance to update the running variance.

In case 3, the unbiased estimate of the batch variance is used to update the running variance. Specifically this is a factor (n-1)/n difference.

It's unclear what is really better. See my question [here](https://discuss.pytorch.org/t/running-mean-variance-biased-or-unbiased-running-var-via-running-mean-or-cur-mean/204486).
